curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -966,16 +993,18 @@ CURVEFS_ERROR FileCacheManager::Flush(bool force, bool toS3) {
                     return;
                 }
                 VLOG(6) << "ChunkCacheManagerPtr count:"
-                        << context->chunkCacheManptr.use_count();
-                // tmp and chunkCacheMap_ has this ChunkCacheManagerPtr,
-                // so count is 2 if count more than 2,
+                        << context->chunkCacheManptr.use_count()
+                        << ",inode:" << inode_
+                        << ", index:" << context->chunkCacheManptr->GetIndex();
+                // tmp、chunkCacheMap_ and context->chunkCacheManptr has this
+                // ChunkCacheManagerPtr, so count is 3 if count more than 3,
                 // this mean someone thread has this ChunkCacheManagerPtr
                 if (iter1->second->IsEmpty() &&
-                  (2 == iter1->second.use_count())) {
+                  (iter1->second.use_count() <= 3)) {
```

Comment:
If the context also holds chunkCacheManptr, it is released here, will there be no problem?

Reply:
if other context holds chunkCacheManptr, use count must more than 3